### PR TITLE
Feature/SOF 5337

### PR DESCRIPTION
**Pull request by tjduigna**

### CODE REVIEW

File: js/test/action.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    - name: Setup NodeJS
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ inputs.node-version }}
+    - name: Install
+      shell: bash -l {0}
+      run: npm install
+    - name: Build
+      shell: bash -l {0}
+      run: npm run build --if-present
```
Comment (timurbazhirov) [inline js/test/action.yml:26]: Per our conversation, we might want to add the "npm transpile" part here too

Comment (tjduigna) [reply js/test/action.yml:26]: I think the `postinstall` step in the `package.json` handles it cleanly, as it is something that may be configured per package.

File: js/publish/action.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ inputs.node-version }}
+        registry-url: https://registry.npmjs.org
+    - name: Install
+      shell: bash -l {0}
+      run: npm install
+    - name: Publish
+      shell: bash -l {0}
+      run: npm publish --access public
```
Comment (tjduigna) [inline js/publish/action.yml:29]: @timurbazhirov Should we parametrize the publish action to allow publishing private NPM packages or do we only plan to publish open source projects to NPM?

File: js/test/action.yml (hunk 1 above)
Comment (timurbazhirov) [reply js/test/action.yml:26]: OK. I wasn't sure whether the `postinstall` is for when the package is installed as a dependency, but looks like it runs after we do `npm install` as well: https://docs.npmjs.com/cli/v8/using-npm/scripts

File: js/publish/action.yml (hunk 2 above)
Comment (timurbazhirov) [reply js/publish/action.yml:29]: We only have public NPM packages so far, the rest of the codebase is in webapp (and or can be sourced directly from GitHub), but it's always good to engineer for being future-proof, I guess. Your call.

Comment (tjduigna) [reply js/publish/action.yml:29]: Ok, it's easy enough to add the branch logic when the use case arises. No sense overcomplicating it for now.

**Review by timurbazhirov — APPROVED** [review]

File: js/publish/action.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    - name: Publish NPM Release
+      shell: bash -l {0}
+      run: npm publish --access public || echo "this version has already been published"
+      env:
+        NODE_AUTH_TOKEN: ${{ inputs.npm-token }}
+    - name: Publish Github Release
+      shell: bash -l {0}
+      run: |
+        git remote set-url origin https://${{ inputs.github-token }}@github.com/${{ github.repository }}.git
+        VERSION=$(cat package.json | grep version | head -1 | awk -F: '{ print $2 }' | sed 's/[",]//g' | tr -d '[[:space:]]')
```
Comment (timurbazhirov) [inline js/publish/action.yml:46]: Maybe worthwhile taking a look at whether there's a native npm tool or a package to get the version out of package.json

Comment (tjduigna) [reply js/publish/action.yml]: Versioning has been revised to follow a date-based versioning scheme and should allow for automated releases any number of times a day.

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**